metpy/calc/thermo.py

```diff
@@ -840,3 +842,180 @@ def relative_humidity_from_specific_humidity(specific_humidity, temperature, pre
+    if lev[1] > lev[0]:
+        level = np.flipud(lev) * lev.units
+        tmp = np.flipud(tmp) * tmp.units
+    else:
+        level = lev
+    # convert to appropriate units and check for meteorologically sound data,
+    # if not, raise warning to the user.
+    tmpk = tmp.to('kelvin')
+    levels = level.to('hPa')
+    if levels[0] <= 10 * units.hPa or levels[0] >= 100000 * units.hPa:
+        warnings.warn('Unexpected value encountered, check units for accuracy')
+    isentlevels = isentlevs.to('kelvin')
+
+    # Make the pressure levels and desired isentropic levels the same shape as temperature
+    levs = np.repeat(np.repeat(levels[:, np.newaxis],
+                     tmpk.shape[-2], axis=1)[:, :, np.newaxis],
+                     tmpk.shape[-1], axis=2)
+    isentlevs2 = np.repeat(np.repeat(isentlevels[:, np.newaxis],
+                           tmpk.shape[-2], axis=1)[:, :, np.newaxis],
+                           tmpk.shape[-1], axis=2)
+
+    # exponent to Poisson's Equation, which is imported above
+    ka = kappa * 1000 * units('g/kg')
+
+    # calculate theta for each point
+    thtalevs = potential_temperature(levs, tmpk)
+
+    # Find log of pressure to implement assumption of linear temperature dependence on
+    # ln(p)
+    pk = np.log(levs.m)
+
+    # Calculations for interpolation routine
+    pok = 1000. ** (ka)
+
+    # index values for each point for the pressure level nearest to the desired theta level
+    minv = np.apply_along_axis(np.searchsorted, 0, thtalevs.m, isentlevs.m)
+    ar = np.arange
+
+    # calculate constants for the interpolation
+    a = (tmpk.m[minv, ar(tmpk.shape[-2]).reshape(-1, 1), ar(tmpk.shape[-1])] -
+         tmpk.m[minv - 1, ar(tmpk.shape[-2]).reshape(-1, 1),
+                ar(tmpk.shape[-1])]) / (pk[minv, ar(tmpk.shape[-2]).reshape(-1, 1),
+                                           ar(tmpk.shape[-1])] -
+                                        pk[minv - 1, ar(tmpk.shape[-2]).reshape(-1, 1),
+                                           ar(tmpk.shape[-1])])
+
+    b = tmpk.m[minv, ar(tmpk.shape[-2]).reshape(-1, 1),
+               ar(tmpk.shape[-1])] - a * pk[minv,
+                                            ar(tmpk.shape[-2]).reshape(-1, 1),
+                                            ar(tmpk.shape[-1])]
+
+    # calculate first guess for interpolation
+    pk1 = (pk[minv, ar(tmpk.shape[-2]).reshape(-1, 1), ar(tmpk.shape[-1])] +
+           0.5 * (pk[minv + 1, ar(tmpk.shape[-2]).reshape(-1, 1), ar(tmpk.shape[-1])] -
+                  pk[minv, ar(tmpk.shape[-2]).reshape(-1, 1), ar(tmpk.shape[-1])]))
+
+    # iterative interpolation useing scipy.optimize.fixed_point and _isen_iter defined above
+    pk2 = so.fixed_point(_isen_iter, pk1, args=(isentlevs2.m, ka, a, b, pok),
+                         xtol=eps, maxiter=max_iters)
+
+    # get back pressure and assign nan for values with pressure greater than 1000 hPa
+    isentprs = np.exp(pk2)
+    isentprs[isentprs > 1000.] = np.nan
```

Comment:
Should we just make it the highest pressure in the input data, thereby eliminating extrapolated points?